tests/test_resolve.py

```diff
@@ -127,7 +127,7 @@ def test_anaconda_nomkl(self):
 
 def test_generate_eq_1():
     # avoid cache from other tests which may have different result
-    r._reduced_index_cache = {}
+    _index, r = get_index_r_1()
```

Comment:
Doesn't belong in this PR

Reply:
If I leave this out of this PR, then the tests fail.

Reply:
that's odd, shouldn't this be failing on `main` too then?

Reply:
It is re-using a module-scope cached `r`, and something (no idea what) in another test sets it up for success or failure before we run here. So these new splits have improved the quality of our test suite.

The `least_duration` algorithm isn't as aggressive as randomizing test order, but it can split each file's tests into multiple runners compared to the naive algorithm which runs the 1st third of the tests in one batch, and the 2nd third in another batch and so on.

Reply:
ah I see, carry on then

Reply:
auto-merge enabled